SimPEG/EM/Static/DC/ProblemDC.py

```diff
@@ -117,6 +167,38 @@ def getSourceTerm(self):
             q[:, i] = src.eval(self)
         return q
 
+    def getSourceTerm(self):
+        """
+        Evaluates the sources, and puts them in matrix form
+
+        :rtype: tuple
+        :return: q (nC or nN, nSrc)
+        """
+
+        Srcs = self.survey.srcList
+
+        if self._formulation == 'EB':
+            n = self.mesh.nN
+            # return NotImplementedError
+
+        elif self._formulation == 'HJ':
+            n = self.mesh.nC
+
+        q = np.zeros((n, len(Srcs)))
+
+        for i, src in enumerate(Srcs):
+            q[:, i] = src.eval(self)
+        return q
+
+    @property
+    def deleteTheseOnModelUpdate(self):
+        toDelete = []
+        if self.sigmaMap is not None or self.rhoMap is not None:
+            toDelete += ['_MeSigma', '_MeSigmaI', '_MfRho', '_MfRhoI']
```

Comment:
these are already on the base class. You could instead do
```
def deleteTheseOnModelUpdate(self):
    toDelete = super(BaseDCProblem, self).deleteTheseOnModelUpdate
    if self.Jmat is not None:
        toDelete += ['Jmat']
    return toDelete
```

(this highlights that only one thing is really being added beyond what is already on the base class)

Reply:
I thought it overwrites full `deleteTheseOnModelUpdate` function. 
Is that what `super` does?

Reply:
https://docs.python.org/2/library/functions.html#super

Reply:
that's cool.